src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -132,21 +132,29 @@ struct CPUStreamsExecutor::Impl {
                                    const StreamCreateType stream_type,
                                    const int concurrency,
                                    const int core_type,
-                                   const int numa_node_id) {
+                                   const int numa_node_id,
+                                   const int max_threads_per_core) {
             _numaNodeId = (_impl->_usedNumaNodes.size() == 1 && _impl->_usedNumaNodes.at(0) == -1)
                               ? -1  // macOS
                               : std::max(0, numa_node_id);
             _socketId = get_socket_by_numa_node(_numaNodeId);
             if (stream_type == STREAM_WITHOUT_PARAM) {
-                _taskArena.reset(new custom::task_arena{concurrency});
+                _taskArena.reset(new custom::task_arena{custom::task_arena::constraints{}
+                                                            .set_max_concurrency(concurrency)
+                                                            .set_max_threads_per_core(max_threads_per_core)});
```

Comment:
As we discussed lets clarify `set_max_threads_per_core` behavior with legacy TBB before merge.

Reply:
For legacy TBB, openvino uses static tbbbind library and set max_threads_per_core to tbbbind directly.

Reply:
@dmitry-gorokhov Please find the detail at CVS-117357